### DISCUSSION TITLE
Debug Adapter for Crazyflie

### DISCUSSION THREAD

**Discussion by LorenzoLamberti94** [bitcraze/discussions#83] [Q&A]
Hi!
I am reaching out since I was trying the VSCode extension for debugging the STM32
as for an older discussion in the forum (https://forum.bitcraze.io/viewtopic.php?t=5250 ) I can't debug the STM as well
The gdb session closes unexpectedly
I would like to ask if anyone tried this VSCode extension and was able to make it work. What should I doublecheck on my setup ?

thanks ! 

![image](https://user-images.githubusercontent.com/43588878/172836713-25352df8-5956-4f6f-a8e4-b99b3a13c137.png)

**Comment by krichardsson** [comment]
HI!

I use the setup described in [documentation](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/openocd_gdb_debugging/) and it works pretty well. I'm running on Ubuntu and not in a VM as mentioned in the forum thread. Are you using a VM and which OS are you running?

**Comment by karlscholz** [comment]
Hey I had a simelar problem, maybe downgrade your Cortex-Debug Extension to 1.2.2, that worked for me. If that doesn't work, try to connect with the CLI Interface of gdb-multiarch. 
https://github.com/orgs/bitcraze/discussions/331

Also here are the updated instructions for STM Debugging.
https://github.com/bitcraze/crazyflie-firmware/blob/master/docs/development/openocd_gdb_debugging.md
